src/components/FlexibleForm/Control.js

```diff
@@ -207,21 +206,11 @@ export const Control = memo(props => {
         wrapper = decorators[decorator];
       }
 
-      // TODO: Should be moved to utils in the future
-      if (decorator && decorator === 'withBooleanActions' && staticControlType === 'KeyValue') {
-        const actions = get(FORMS_SETTINGS, [ENTITY_KEYS.MAPPING_PROFILES, 'DECORATORS', 'BOOLEAN_ACTIONS'], []);
-        const newValue = actions.find(item => item.value === attrs?.value)?.label;
-
-        attrs = {
-          ...attrs,
-          value: newValue ? (<FormattedMessage id={newValue} />) : newValue,
-        };
-      }
-
       attrs = {
         ...attrs,
         component: wrapper || control,
         WrappedComponent: wrapper ? control : null,
+        value: getBooleanActionsValue(attrs?.value) || noValueComponent,
```

Comment:
What happens if the decorator is any other, not the withBooleanActions?

Reply:
This function converts value only in case when it's Boolean Action Value, for others it returns value as is

Reply:
In this case, maybe it will be better to rename `getBooleanActionsValue` method if it returns a value for any type of decorator not just for `withBooleanActions`?

Reply:
If you have a good name for that function, please advice.